### DISCUSSION TITLE
Hooks with redirect break during build with adapter-node after @sveltejs/kit next.433

### DISCUSSION THREAD

**Discussion by fmaclen** [sveltejs/kit#6325] [Q&A] [answered]
I'm not entirely sure if this is a bug or my implementation is mistaken.

Here's what my code looks like:

```javascript
// src/hooks.ts

export const handle: Handle = async ({ event, resolve }) => {
	if (shouldRedirect) {
		return Response.redirect(`${event.url.origin}/some-route`, 307);
	} else {
		return await resolve(event);
	}
};
```

When I run `npm run build` using `adapter-node@1.0.0-next.87` the last few lines of the logs look like this:

```
.svelte-kit/output/server/chunks/hooks.js                                     0.52 KiB
  307 / -> http://sveltekit-prerender/some-route
  307 /route-one -> http://sveltekit-prerender/some-route
  307 /route-two -> http://sveltekit-prerender/some-route
  307 /route-three -> http://sveltekit-prerender/some-route
```

Then, running `npm run preview` (when the redirect condition is matched) my browser is redirected to `http://sveltekit-prerender/some-route` which is obviously not a valid URL.

The code above was working fine up until `@sveltejs/kit@1.0.0-next.432` so I'm guessing the behavior changed on https://github.com/sveltejs/kit/pull/6179

Code works as expected when `npm run dev`.

**Edit** — Here's a repro: https://stackblitz.com/edit/sveltejs-kit-template-default-bbeykp?file=src%2Froutes%2F%2Bpage.js&file=src%2Fhooks.js&terminal=dev

**Comment by fmaclen** [comment, marked as answer]
After further investigation I was able to solve the issue by disabling prerender in [`svelte.config.js`](https://kit.svelte.dev/docs/configuration#prerender) globally.

I'm not sure if there's a way to disable prerendering for specific conditions in `hooks.ts` but disabling it globally works in my case.

**Comment by fmaclen** [comment]
@tcc-sejohnson can you confirm if this is the expected behavior or should I open an issue?

**Reply by mustofa-id** [reply]
There is already an open [issue](https://github.com/sveltejs/kit/issues/5687) and I give temp [workaround](https://github.com/sveltejs/kit/issues/5687#issuecomment-1227188153).
